apps/server/src/modules/group/repo/group.repo.ts

```diff
@@ -205,4 +163,39 @@ export class GroupRepo extends BaseDomainObjectRepo<Group, GroupEntity> {
 
 		return domainObjects;
 	}
+
+	private buildPipeline(skip?: number, limit?: number, nameQuery?: string): unknown[] {
+		let nameQueryMatch = {};
+		const pipeline: unknown[] = [];
+
+		if (nameQuery && StringValidator.isNotEmptyString(nameQuery, true)) {
+			const escapedName = nameQuery.replace(MongoPatterns.REGEX_MONGO_LANGUAGE_PATTERN_WHITELIST, '').trim();
+			if (StringValidator.isNotEmptyString(escapedName, true)) {
```

Comment:
Do you need nested if's here? Think about it

Reply:
unfortunately i think this makes sense... we check if nameQuery is given and notEmpty or consisting of only whitespaces, we then remove all non-language characters except numbers, "-" and whitespaces. After this removal we now could have an empty string or one that consists only of whitespaces, so we should check for that. And there is your nested if...